physics/discrete_trajectory.hpp

```diff
@@ -174,6 +182,22 @@ class DiscreteTrajectory : public Forkable<DiscreteTrajectory<Frame>,
   std::int64_t timeline_size() const override;
 
  private:
+  struct Downsampling {
```

Comment:
I'd like to have a 2-argument constructor taking `max_dense_intervals` and `tolerance`.  Then the last two members may be made `const`.

In fact this should be a bona-fide class with actual code, see my comments below.

Reply:
Done.